chromium_src/chrome/browser/autocomplete/chrome_autocomplete_provider_client.cc

```diff
@@ -71,11 +71,16 @@ void ChromeAutocompleteProviderClient::OpenLeo(const std::u16string& query) {
       sidebar::SidebarItem::BuiltInItemType::kChatUI);
 
   // Send the query to the AIChat's backend.
-  ai_chat::mojom::ConversationTurn turn = {
-      ai_chat::mojom::CharacterType::HUMAN, ai_chat::mojom::ActionType::QUERY,
-      ai_chat::mojom::ConversationTurnVisibility::VISIBLE,
-      base::UTF16ToUTF8(query), std::nullopt};
+  ai_chat::mojom::ConversationTurnPtr turn =
+      ai_chat::mojom::ConversationTurn::New();
```

Comment:
I was wondering what best practice is here. As a counterpoint, the more params a struct has, the harder it is to see which constructor params do what, and IMO it becomes easier to introduce bugs with wrong constructor param order. Many params are optional, so we end up with a sprinkling of `std::nullopt` in amongst the parameters.

Reply:
Comments can be used inline to say what it is.
Mainly what I want to avoid is we accidentally using some default value that shouldn't be used because we miss a spot.

Reply:
Especially this file might be easier to miss since it's not in ai_chat component.